### DISCUSSION TITLE
Reading data files with ctk is pure frustration

### DISCUSSION THREAD

**Discussion by Heroeskeeper** [TomSchimansky/CustomTkinter#2020] [General]
I have written Python script to read a data file using a given structure reference and index file and it works fine running from command prompt. 
As soon as I try to apply a decent UI like ctk it all turns to crap. Now the data interpretation is corrupted. I simply copy pasted the extraction method into the ctk script. 
My 2 front teeth for a decent python UI that can read data files as well as a bad UI does - a UI shouldn't affect the way a script reads a file.
Using classes makes things even worse, - this is supposed to make things easier!
Maybe I'm just green behind the ears. but logically, CTK is broken.

**Comment by ElectricCandlelight** [comment]
You've not really explained what you want to do or provided a minimum example.

I've no idea what you even mean by "but logically, CTK is broken."  It has worked fine for all of the projects I've used it on.

**Reply by Heroeskeeper** [reply]
Hi, apologies for my rant.
The code below: 
```
import customtkinter as ctk

ctk.set_appearance_mode("dark")

word = ""

class Frame1(ctk.CTkFrame):
    def __init__(self, master, word):
        super().__init__(master)
        self.word = word

        self.name = ctk.CTkLabel(self, text="Name", fg_color="transparent")
        self.name.grid(row=1, column=0, padx=10, pady=10, sticky="w")

        self.name_x = ctk.CTkEntry(self, width = 300, placeholder_text="")
        self.name_x.grid(row=1, column=1, padx=10, pady=10, sticky="e")
        self.name_x.insert("end", self.word)
   
class App(ctk.CTk):
    def __init__(self):
        super().__init__()

        self.title("CRD Master")
        self.geometry("1000x650")
        self.grid_columnconfigure(0, weight=1)
        self.grid_rowconfigure(0, weight=1)

        self.acc_no = ctk.CTkEntry(self, width = 60, placeholder_text="Acc. no.")
        self.acc_no.grid(row=0, column=0, padx=10, pady=10, sticky="w")

        self.button = ctk.CTkButton(self, width = 60, text="get", command=self.button_callback)
        self.button.grid(row=0, column=0, padx=10, pady=10, sticky="e")

        frame = Frame1(self, word)
        frame.grid(row=1, column=0, padx=10, pady=(10, 0), sticky="nsw")

    def button_callback(self):
        word = "yay" 
        # Frame1.frame(word)
        # print(word)

app=App()
app.mainloop()
```
All I'm trying to do in this test code is display the word 'yay' in the ctkentry widget thats inside a ctkframe class.
There are no resources on the internet that shows how to use a ctkentry widget thats inside a ctkframe class to show information. It must surely be possible? Please help.

**Comment by domhnallmorr** [comment]
Do you want "yay" to appear as the default text in the ctkentry widget, or do you only want it to appear after the user has clicked the "get" button?

**Reply by Heroeskeeper** [reply]
After the user has clicked the 'get' button.

**Comment by domhnallmorr** [comment]
Ok so here is one solution that requires a small change of your code;

Note the following line:

self.frame = Frame1(self, word)

I don't know how experienced with python you are. If not very, then I would recommend to have a look at some material out there on classes and objects/object orientated programming to understand why that works.

Hope this helps!

```
import customtkinter as ctk

ctk.set_appearance_mode("dark")

word = ""

class Frame1(ctk.CTkFrame):
    def __init__(self, master, word):
        super().__init__(master)
        self.word = word

        self.name = ctk.CTkLabel(self, text="Name", fg_color="transparent")
        self.name.grid(row=1, column=0, padx=10, pady=10, sticky="w")

        self.name_x = ctk.CTkEntry(self, width = 300, placeholder_text="")
        self.name_x.grid(row=1, column=1, padx=10, pady=10, sticky="e")
        self.name_x.insert("end", self.word)
   
class App(ctk.CTk):
    def __init__(self):
        super().__init__()

        self.title("CRD Master")
        self.geometry("1000x650")
        self.grid_columnconfigure(0, weight=1)
        self.grid_rowconfigure(0, weight=1)

        self.acc_no = ctk.CTkEntry(self, width = 60, placeholder_text="Acc. no.")
        self.acc_no.grid(row=0, column=0, padx=10, pady=10, sticky="w")

        self.button = ctk.CTkButton(self, width = 60, text="get", command=self.button_callback)
        self.button.grid(row=0, column=0, padx=10, pady=10, sticky="e")

        self.frame = Frame1(self, word)
        self.frame.grid(row=1, column=0, padx=10, pady=(10, 0), sticky="nsw")

    def button_callback(self):
        word = "yay" 
        self.frame.name_x.insert(0, word)

app=App()
app.mainloop()
```

**Reply by Heroeskeeper** [reply]
domhnallmorr you are a gentleman, thank you. I am a python newbie and will definitely take your advice. It works and I understand it better now.
